### PR TITLE
delete test feature

### PR DESCRIPTION
**Pull request by ritika05**
**Fixed Issues**

#21191, #21193, #21195

**Files Changed**

* TestController.cs - Added server side delete test code
* ITestRepository.cs - Added server side delete test code
* TestRepository.cs - Added server side delete test code
* delete-test-dialog.component.ts - Added client side delete test code
* delete-test-dialog.html - Added delete validation dialog
* tests.service.ts - Added delete test API call
* test-dashboard.component.ts - Added DeleteTestDialogComponent's component Instance
* test-dashboard.html - Passed test object as parameter in deleteTest
* TestsRepositoryTest.cs -  Added unit testing for delete test code

**Checks**

- [x] Naming Conventions
- [x] Server side code comments (proper English, grammar and no spelling mistakes)
- [x] Client side code comments (proper English, grammar and no spelling mistakes)
- [x] Unused variables, methods, blank spaces and name spaces. 
- [x] Optimal Code and code formatting
- [x] Commit History is proper, no merge is done. 
- [x] Commit/pull request messages should be proper to justify your feature
- [x] All test cases are executed provided by tester.

**Known Issues**

N/A

**Comments**

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs [hunk 1]
```diff
@@ -105,5 +105,20 @@ public TestsRepository(TrappistDbContext dbContext, IGlobalUtil util)
         {
             return await _dbContext.Test.AnyAsync(x => x.Id == id);
         }
+
+        public async Task<bool> IsAttendeeExistAsync(int id)
+        {
+            Test test = await _dbContext.Test.Include(x => x.TestAttendees).FirstOrDefaultAsync(x => x.Id == id);
+            if (test.TestAttendees.Count != 0)
```
Comment (iroshni) [inline Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs]: Count can be replaced with Any()

Comment (iroshni) [inline Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs]: and the if condition can be directly used as return statement.

File: Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs [hunk 2]
```diff
@@ -106,11 +106,36 @@ public TestsRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
...
+            await _testRepository.UpdateTestNameAsync(test.Id, testNameToUpdate);
             var TestName = "MCKV";
             Assert.True(_trappistDbContext.Test.Count(x => x.TestName == TestName) == 1);
         }
 
+        /// <summary>
+        /// Check if any test attendee exist for a particular test
+        /// </summary>
+        [Fact]
+        public async Task IsAttendeeExistsAsync()
```
Comment (iroshni) [inline Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs]: Missing scenarios of delete test cases. identify and write them. All the conditions you have checked should be covered.

File: Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs [hunk 3]
```diff
@@ -120,5 +160,17 @@ private DomainModel.Models.Test.Test CreateTest(string testName)
             };
             return test;
         }
+
+        private DomainModel.Models.TestConduct.TestAttendees TestAttendee()
+        {
+            var test = CreateTest("Quantitative");
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Test/Tests/TestsRepositoryTest.cs]: MAJOR Remove this unused 'test' local variable. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1481)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**